src/commands/classes/Contest.command.ts

```diff
@@ -108,21 +235,116 @@ export default class ContestCommand extends Command {
   //top 3 get move to unused entries folder to be reshuffled once all entries have been gone through
   public end(args: Array<string>, message: Message): void{ //
+    if(contestData.contestActive){
+      console.log("---Ending Contest---");
+      let reaction_numbers = ["\u0030\u20E3","\u0031\u20E3","\u0032\u20E3","\u0033\u20E3","\u0034\u20E3","\u0035\u20E3", "\u0036\u20E3","\u0037\u20E3","\u0038\u20E3","\u0039\u20E3"];
+      let files = contestData.entries;
+      console.log("Determining contest channel");
+      let contestChannel = message.guild.channels.cache.find((channel) => channel.name === contestData.contestChannelName) as TextChannel;
+      console.log("Successful");
+      
+      console.log("Determining Contest origin message");
+      contestChannel.messages.fetch(contestData.messageId).then(contestMessage => {
+        //console.log("ContestMessage ID: "+contestMessage.id+"\t\t"+contestMessage.toString());
+        
+        let reactions = contestMessage.reactions.cache;
+        console.log("Successful");
+        let reacEmoji = [];
+        let reacCount = [];
+        for (let reaction of reactions)
+        {
+          reacEmoji.push(reaction[1].emoji);
+          console.log(reaction);
+          reacCount.push(reaction[1].count);
+          console.log("reacCount:  " + reacCount);
+
+        }
+
+        console.log("Determining winners");
+        let winners = [];
+        while(winners.length < 3){
+          let max = 0;
+          for(let x of reacCount){
+            max = x > max ? x : max;
+          }
+          
+          while(reacCount.includes(max)){
+            let winner = {};
+            let x = reacCount.indexOf(max);
+            winner["emoji"] = reacEmoji[x];
+            console.log("saving reacCount: "+reacCount[x]);
+            winner["count"] = reacCount[x];
+            winner["file"] = files[x];
+            files.splice(x,1);
+            reacEmoji.splice(x,1);
+            reacCount.splice(x,1);
+            winners.push(winner);
+          }
+
+        }
+        console.log("Successful");
+
+        console.log("Creating announcement message");
+        //console.log(winners);
+        let announcement = "\nThe contest has ended and our winners are: "
+        for(let x=0; x<winners.length; x++){
+          announcement += `\n${winners[x].count} votes:   [${winners[x].file.name}]  `;
+          let urls = `\n${winners[x].file.url} `;
+          while( (x+1 < winners.length) && (winners[x].count == winners[x+1].count)  ){
+            x++;
+            announcement += `and  [${winners[x].file.name}]  `;
+            urls += `\n\n${winners[x].file.url} `;
+          }
+          announcement += urls+"\n";
+      
+        }
+        announcement += "\n :tada:CONGLATIONS:tada: \n\n Remember to upload your files before the next contest!";
+        console.log('Successful');
+        
+        for(let x=0; x<winners.length; x++){
+          contestData.pastEntries.push(winners[x].file);
```

Comment:
Also, there should be a max pastEntries that get pushed out as more pastEntries exist. Or, when we start, make sure there is enough in the file list. If not, remove `count - filelist.length` from pastEntries.